src/components/segmentedControl/index.tsx

```diff
@@ -106,8 +106,8 @@ const SegmentedControl = (props: SegmentedControlProps) => {
 
   const onSegmentPress = useCallback((index: number) => {
     if (selectedSegment !== index) {
-      onChangeIndex?.(index);
       updateSelectedSegment(index);
+      setTimeout(() => onChangeIndex?.(index), 400);  
```

Comment:
Other than delaying the callback, I'd consider also to throttle the calls. 
For instance, if the user press quickly on multiple different segments, maybe we can throttles all of these calls and trigger only the last one. 

Where are you checking the performance of this component? It's mostly noticeable in our SectionWheelPicker example screen. 

Another thing worth checking (can be in another PR) is to move to reanimated2, other than simplifying the code it can also help with performance.

Reply:
I tried to wrap the callback with lodash throttle but couldn't see any change.. also, I couldn't see the throttle effect in other places we use it (PickerModal for example), let me know if it's only me

I'm checking it in the SectionWheelPickerScreen as well :)

I have a draft PR on moving this component to reanimated2, I'll publish it after this PR :)